### ISSUE TITLE
Consider new front end frameworks / libraries

### ISSUE THREAD

**Issue by alexsielicki** [sandialabs/slycat#708]
We are currently using Knockout, but should reconsider other frameworks and libraries out there, like Angular and React.

**Comment by alexsielicki** [comment]
To get things started, here's a performance comparison between React and Angular, although from January 2015:
https://www.codementor.io/reactjs/tutorial/reactjs-vs-angular-js-performance-comparison-knockout
Interestingly, React turns out to be even faster than direct DOM manipulation. Perhaps we can find a more recent similar comparison.

**Comment by alexsielicki** [comment]
More benchmarks, this time from 2016:
https://auth0.com/blog/updated-and-improved-more-benchmarks-virtual-dom-vs-angular-12-vs-mithril-js-vs-the-rest/

**Comment by Mletter1** [comment]
I like that "React is black-magic" in terms of speed :D 💯

**Comment by alexsielicki** [comment]
I've heard good things about Redux, especially when used with React. Let's take a deeper look at it to manage Slycat's state:
http://redux.js.org/docs/basics/UsageWithReact.html

**Comment by alexsielicki** [comment]
This is also worth checking out:
http://stateofjs.com

TL;DR
- ES6 is the new standard.
- React is great. Vue is the new up-and-comer.
- Redux is the top dog in state management.
- Webpack seems to be the leader in build tools.

**Comment by alexsielicki** [comment]
We have adapted React, Redux, ES6, and Webpack, so I'm closing this.
